sv_inline.h

```diff
@@ -903,7 +903,7 @@ Perl_SvPV_helper(pTHX_
             return SvPVX_mutable(sv);
         }
         else if(return_flags & SV_CONST_RETURN) {
-            return (char *) SvPVX_const(sv);
+            return (char *)(size_t)SvPVX_const(sv);
```

Comment:
I don't really like this one at all. It's making the code worse just to get rid of a warning that isn't helpful in this particular case.

Reply:
Please be constructive.

Why do you write that I'm making the code worse? The cast that I inserted does not affect the generated code.

SvPVX_const() yields a const pointer while Perl_SvPV_helper() has return type `char *` so I think there is a fundamental issue at hand. I don't see how casting away constness can be prevented other than by converting Perl_SvPV_helper() from a function into a macro?

Reply:
> Why do you write that I'm making the code worse? 

Because casting a pointer to an integer and then back to a pointer is really confusing. And ultimately the human reader is more important to me than warning that advices against code that is both correct and standard conformant .

Reply:
This function seems to assume `SvPVX_mutable`, `SvPVX` and `SvPVX_const` do different things, when in fact they're only different in type (and sometimes lvalueness). I suspect the better solution is to just always return SvPVX, unless we want to make them do different things in the future.

Reply:
> > Why do you write that I'm making the code worse?
> 
> Because casting a pointer to an integer and then back to a pointer is really confusing. And ultimately the human reader is more important to me than warning that advices against code that is both correct and standard conformant .

I want to enable both -Werror and -Wcast-qual in the Net-SNMP project but I cannot do that because of the warnings triggered by the Perl headers. Suggestions for alternatives to the Perl code changes I proposed are welcome.

Reply:
Well, you have now learned why using `-Werror` is a bad idea. Compilers introduce new warnings all the time, usually they're harmless, but with `-Werror` they will break your build. And the worst part is that those warnings can be triggered by third party code you have no control over, because C doesn't have a module system.

Reply:
It's much the same reason why we don't recommend `use warnings FATAL => ':all';`

Reply:
`-Wcast-qual` has been enabled in the Net-SNMP code base to find the root cause of a certain class of crashes and also to prevent that similar crashes get reintroduced in the future.